core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ExpressionEquivalence.java

```diff
@@ -262,9 +266,7 @@ public int compare(RowExpression left, RowExpression right)
                     }
                 }
 
-                // value is some random type (say regex), so we just randomly choose a greater value
-                // todo: support all known type
-                return -1;
+                throw new IllegalArgumentException("Cannot compare " + leftConstant + " of type " + leftConstant.getType().getJavaType() + " and " + rightConstant + " of type " + rightConstant.getType().getJavaType());
```

Comment:
- use string formatting
- printing `leftConstant.getType()` gives more information than printing `leftConstant.getType().getJavaType`, and it's shorter, so let's do that